src/mscorlib/src/System/Environment.cs

```diff
@@ -536,23 +537,33 @@ private static void ValidateTarget(EnvironmentVariableTarget target)
 
         private static string GetEnvironmentVariableCore(string variable)
         {
-            StringBuilder sb = StringBuilderCache.Acquire(128); // A somewhat reasonable default size
-            int requiredSize = Win32Native.GetEnvironmentVariable(variable, sb, sb.Capacity);
+            Span<char> buffer = stackalloc char[128]; // A somewhat reasonable default size
```

Comment:
is it reasonable? since this is now a stack allocation, maybe it's cheap to be more generous? although, this is probably fine for most variables except path, include, libpath, etc.

Reply:
@stephentoub do we have a rule of thumb for reasonable stackalloc max? In the CoreFX code I see mostly a few bytes, a cutoff of 100, 200, one of 256 is the largest.

Reply:
I had the same question. 128 chars (256 bytes) seemed like a reasonable limit... but I could see going a little higher. Not sure how far we want to push it, though.

Reply:
We don't have a hard and fast rule, but we've generally stayed at <= 256 bytes.